scalameta/parsers/js/src/main/scala/scala/meta/parsers/JSFacade.scala

```diff
@@ -0,0 +1,80 @@
+package scala.meta
+package parsers
+
+import scala.scalajs.js
+import js.JSConverters._
+import js.annotation._
+
+object JSFacade {
+
+  // https://stackoverflow.com/a/36573183/846273
+  private[this] def mergeJSObjects(objs: js.Dynamic*): js.Dynamic = {
+    val result = js.Dictionary.empty[Any]
+    for (source <- objs) {
+      for ((key, value) <- source.asInstanceOf[js.Dictionary[Any]])
+        result(key) = value
+    }
+    result.asInstanceOf[js.Dynamic]
+  }
+
+  // https://github.com/scala-js/scala-js/issues/2170#issuecomment-176795604
+  @js.native
+  private[this] sealed trait JSLong extends js.Any
+  implicit private[this] class LongJSOps(val x: Long) extends AnyVal {
+    def toJSLong: JSLong = {
+      if (x >= 0) (x & ((1L << 53) - 1)).toDouble
+      else -((-x) & ((1L << 53) - 1)).toDouble
+    }.asInstanceOf[JSLong]
+  }
+
+  private[this] def toNode(t: Tree): js.Dynamic = {
+    val base = js.Dynamic.literal(
+      "type" -> t.productPrefix,
+      "children" -> t.children.map(toNode).toJSArray,
+      "pos" -> js.Dynamic.literal(
+        "start" -> t.pos.start.offset,
+        "end" -> t.pos.end.offset
+      )
+    )
+
+    def v[A](a: A): js.Dynamic =
+      js.Dynamic.literal("value" -> a.asInstanceOf[js.Any])
+
+    // FIXME(gabro)
+    // This is to make the facade immediately useful by showing the trees' values
+    // Ideally `.toMap` on Tree will take care of adding the '.value' attribute.
+    // See https://github.com/scalameta/scalameta/issues/948
+    val value = t match {
+      case t: Lit.Int => v(t.value)
+      case t: Lit.Double => v(t.value)
+      case t: Lit.Float => v(t.value)
+      case t: Lit.Byte => v(t.value)
+      case t: Lit.Short => v(t.value)
+      case t: Lit.Char => v(t.value.toString)
+      case t: Lit.Long => v(t.value.toJSLong)
+      case t: Lit.Boolean => v(t.value)
+      case t: Lit.String => v(t.value)
+      case t: Lit.Symbol => v(t.value.name)
+      case t: Name => v(t.value)
+      case _ => js.Dynamic.literal()
+    }
+
+    mergeJSObjects(base, value)
+  }
+
+  private[this] def parse[A <: Tree: Parse](code: String): js.Dictionary[Any] =
+    code.parse[A] match {
```

Comment:
Note. This will Scala211 by default, which does not support trailing commas and some other features. We are actually planning to drop dialects for parsing/pretty-printing so this may not be a problem soon.